garden_ai/gardens.py

```diff
@@ -116,16 +112,47 @@ class Config:
     version: str = "0.0.1"
     uuid: UUID = Field(default_factory=uuid4, allow_mutation=False)
     pipeline_ids: List[UUID] = Field(default_factory=list)
+    pipeline_aliases: Dict[str, str] = Field(default_factory=dict)
+    _pipelines: List[RegisteredPipeline] = PrivateAttr(default_factory=list)
+    _env_vars: Dict[str, str] = PrivateAttr(default_factory=dict)
 
     @validator("year")
     def valid_year(cls, year):
         if len(str(year)) != 4:
             raise ValueError("year must be formatted `YYYY`")
         return str(year)
 
-    def collect_pipelines(self) -> List[RegisteredPipeline]:
-        """Collect the full ``RegisteredPipeline`` objects referred to by this garden's pipeline_ids."""
-        from .local_data import get_local_pipeline_by_uuid
+    @property
+    def pipelines(self) -> List[RegisteredPipeline]:
+        """Read-only list of the pipelines registered to a garden.
+
+        Note that these pipelines can also be accessed from the garden as attributes,
+        according to their short_name -- see also ``rename_pipeline``.
+        """
+        if not self._pipelines:
+            self._pipelines = self._collect_pipelines()
+        return self._pipelines
```

Comment:
Yeah, that was also a question mark for me when I was writing it, ended up just picking this approach on vibes -- it would indeed become out of sync if someone had a running notebook/interpreter and also was updating the state of the garden from the cli, but I felt like that's less magical than it checking local data every time the attribute is accessed. Maybe a `garden.refresh()` (rehydrate, to steal a term?) helper? or just making `_collect_pipelines()` back into `collect_pipelines()`, so there's a little more control?

Reply:
I don't love an explicit `refresh()`. I think I would rather they just create a fresh Garden object in that case.

I think the crux here for me is, can the property only become out of date because of changes external to the notebook/interpreter session? In that case it's not crazy that if they're doing something in a totally separate process they might have to re-create the Garden instance. But if within the notebook session the user can add a pipeline to their Garden and not see it reflected in `pipelines`, then I think that's a blocker.

Reply:
Yeah I think not having a `client.add_pipeline` method was probably an oversight. I'm probably satisfied with the potential cache issue as long as we make sure that new method updates the cached property whenever we get around to adding it.

Reply:
agreed - in fairness, i think we actually did have an `add_pipeline` method on gardens that I took out when we made the split between `Pipelines`/`RegisteredPipelines`, but really should have made that a client method instead of just removing it altogether. That's outside the scope of this PR though, you think?